packages/web-components/fast-router/src/links.ts

```diff
@@ -0,0 +1,88 @@
+import { Navigation } from "./navigation";
+
+interface AnchorEventInfo {
+    shouldHandleEvent: boolean;
+    href: string | null;
+    anchor: Element | null;
+}
+
+export interface LinkHandler {
+    connect(): void;
+    disconnect(): void;
+}
+
+export class DefaultLinkHandler implements LinkHandler {
+    private handler = (e: MouseEvent) => {
+        let { shouldHandleEvent, href } = this.getEventInfo(e);
+
+        if (shouldHandleEvent) {
+            e.preventDefault();
```

Comment:
Should we explicitly check for navigation events before preventing default? As such, is it possible that we accidentally catch a button submit event and preventDefault on that?